### ISSUE TITLE
gitk: couldn't execute "git": file name too long

### ISSUE THREAD

**Issue by anb0s** [git-for-windows/git#2122]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
64Bit
```
$ git --version --build-options

git version 2.20.1.windows.1
cpu: x86_64
built from commit: 7c9fbc07db0e2939b36095df45864b8cda19b64f 
sizeof-long: 4
sizeof-size_t: 8

```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

** insert your machine's response here **

Microsoft Windows [Version 6.1.7601]

```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
Editor Option: VIM
Custom Editor Path:
Path Option: BashOnly
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: CRLFCommitAsIs
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Disabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

NO

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
gitk
```
 - What did you expect to occur after running these commands?

gitk window is opened without errors

 - What actually happened instead?

It open a window, but shows error and file changes are not shown

![image](https://user-images.githubusercontent.com/95811/54187808-a7b27700-44ae-11e9-8123-a9287e44005f.png)

**Comment by anb0s** [comment]
same issue with
```
git version 2.21.0.windows.1
cpu: x86_64
built from commit: 2481c4cbe949856f270a3ee80c802f5dd89381aa
sizeof-long: 4
sizeof-size_t: 8
```

**Comment by anb0s** [comment]
If it's not reproducible for you, how can i help to investigate the root couse? Thanks!

**Comment by dscho** [comment]
Isn't this precisely the same bug which https://github.com/git-for-windows/git/pull/2053 tries to fix (and which was reported as https://github.com/git-for-windows/git/issues/1987)?

**Comment by anb0s** [comment]
> Isn't this precisely the same bug which #2053 tries to fix (and which was reported as #1987)?
Yes, it looks like same issue. Sorry i've searched only for opened issues. The #1987 is closed, but the issue is not solved. Should the #1987 be re-opened and this one closed? Thanks!

**Comment by dscho** [comment]
> Sorry i've searched only for opened issues.

We explicitly ask to search also open issues: "I was not able to find an open or closed issue matching what I'm seeing" (and [you put a checkmark there](https://github.com/git-for-windows/git/issues/2122#issue-419875707)). And #1987 should not be re-opened because we should not distribute discussions across a ticket and a PR.

BTW if you want to help, please feel free to jump over to #2053 and assist in fixing the issue.
